src/compack.F90

```diff
@@ -28,416 +28,416 @@
+        !  data section.
+        !
+        if (nbitsd.ne.0) then
+           !   pack first original value
+           if (ival1.ge.0) then
+              call g2_sbytec(cpack,ival1,iofst,nbitsd)
+              iofst=iofst+nbitsd
+           else
+              call g2_sbytec(cpack,1,iofst,1)
+              iofst=iofst+1
+              call g2_sbytec(cpack,iabs(ival1),iofst,nbitsd-1)
+              iofst=iofst+nbitsd-1
            endif
-           !
-           !  subtract min value from spatial diff field
-           !
-           isd=idrstmpl(17)+1
-           minsd=minval(ifld(isd:ndpts))
-           do j=isd,ndpts
-              ifld(j)=ifld(j)-minsd
-           enddo
-           !
-           !   find num of bits need to store minsd and add 1 extra bit
-           !   to indicate sign
-           !
-           nbitsd=i1log2(abs(minsd))+1
-           !
-           !   find num of bits need to store ifld(1) ( and ifld(2)
-           !   if using 2nd order differencing )
-           !
-           maxorig=ival1
-           if (idrstmpl(17).eq.2.and.ival2.gt.ival1) maxorig=ival2
-           nbitorig=i1log2(maxorig)+1
-           if (nbitorig.gt.nbitsd) nbitsd=nbitorig
-           !   increase number of bits to even multiple of 8 ( octet )
-           if (mod(nbitsd,8).ne.0) nbitsd=nbitsd+(8-mod(nbitsd,8))
-           !
-           !  Store extra spatial differencing info into the packed
-           !  data section.
-           !
-           if (nbitsd.ne.0) then
-              !   pack first original value
-              if (ival1.ge.0) then
-                 call g2_sbytec(cpack,ival1,iofst,nbitsd)
-                 iofst=iofst+nbitsd
-              else
-                 call g2_sbytec(cpack,1,iofst,1)
-                 iofst=iofst+1
-                 call g2_sbytec(cpack,iabs(ival1),iofst,nbitsd-1)
-                 iofst=iofst+nbitsd-1
-              endif
-              if (idrstmpl(17).eq.2) then
-               !  pack second original value
-                 if (ival2.ge.0) then
-                    call g2_sbytec(cpack,ival2,iofst,nbitsd)
-                    iofst=iofst+nbitsd
-                 else
-                    call g2_sbytec(cpack,1,iofst,1)
-                    iofst=iofst+1
-                    call g2_sbytec(cpack,iabs(ival2),iofst,nbitsd-1)
-                    iofst=iofst+nbitsd-1
-                 endif
-              endif
-              !  pack overall min of spatial differences
-              if (minsd.ge.0) then
-                 call g2_sbytec(cpack,minsd,iofst,nbitsd)
+           if (idrstmpl(17).eq.2) then
+              !  pack second original value
+              if (ival2.ge.0) then
+                 call g2_sbytec(cpack,ival2,iofst,nbitsd)
                  iofst=iofst+nbitsd
               else
                  call g2_sbytec(cpack,1,iofst,1)
                  iofst=iofst+1
-                 call g2_sbytec(cpack,iabs(minsd),iofst,nbitsd-1)
+                 call g2_sbytec(cpack,iabs(ival2),iofst,nbitsd-1)
```

Comment:
Why this change?

Reply:
It's only an apparent change because the code looks so similar in two places and is confusing diff. If you look at the line nubmers you see that line 185 is supposed to have ival3 and line 196 minsd, and both do.

Reply:
Wow github, not cool. Thx. I noticed the same thing happening later on, and didn't comment in case it was something like this.

Reply:
This is the auto system comparison issue.